RetailCoder.VBE/Inspections/QuickFixes/AssignedByValParameterMakeLocalCopyQuickFix.cs

```diff
@@ -97,13 +97,42 @@ private void ReplaceAssignedByValParameterReferences(IModuleRewriter rewriter, s
             }
         }
 
+
         private void InsertLocalVariableDeclarationAndAssignment(IModuleRewriter rewriter, string localIdentifier)
         {
-            var content = Tokens.Dim + " " + localIdentifier + " " + Tokens.As + " " + _target.AsTypeName + Environment.NewLine
-                + (_target.AsTypeDeclaration is ClassModuleDeclaration ? Tokens.Set + " " : string.Empty)
-                + localIdentifier + " = " + _target.IdentifierName;
+            var content = Tokens.Dim + " " + localIdentifier + " " + Tokens.As + " " + _target.AsTypeName + Environment.NewLine;
+            if (IsBaseTypeContext(_target))
+            {
+                content = content + localIdentifier + " = " + _target.IdentifierName;
+            }
+            else 
+            {
+                //All we can know is that it is not a Base type.  Let VBA determine
+                //the right way to assign the parameter.  The user can simplify it later.
+                string insertIsObjectCheck =
+@"If(IsObject({1})) Then
```

Comment:
@comintern absolutely - basically the idea is to have an abstraction around the presence of that attribute on a member of the type, so that we can easily tell that an object can be legally assigned without the `Set` keyword. I like `IsEnumerable`, but `ImplicitlyAssignable` seems a bit vague. Although, with proper XML-doc, it's still a major improvement...